rs-matter/src/mdns/proto.rs

```diff
@@ -359,14 +364,14 @@ impl<'a> Service<'a> {
     ) -> Result<(), PushError> {
         answer.push((
             Self::dns_sd_fqdn(false).unwrap(),
-            Class::In,
+            dns_class_with_flush(Class::In),
             ttl_sec,
             Ptr::new(self.service_type_fqdn(false).unwrap()),
```

Comment:
This is a PTR record, yet and contrary to the PR name, you are cache-flushing it. Here specifically, I think that's not OK, because we are not the authority of "dns_sd_fqdn" (i.e. to the answer "what service types do you have" we won't be the only ones who would respond)

Reply:
Oh, good catch -- you're right. I've fixed this one and the other instances you've spotted, too.